rectools/metrics/intersection.py

```diff
@@ -0,0 +1,71 @@
+from typing import Optional
```

Comment:
For implementing a new metric apart from the metric itself it is also necessary to implement some other pieces of code to make it work in the pipelines. This is a reference PR for you: https://github.com/MobileTeleSystems/RecTools/pull/81/files
What is missing now:
- `calc_intersection_metrics` in `intersection.py`
- `calc_intersection_metrics` call in `scoring.py`
- new metric added to `metrics.__init__.py`
- new metric added to `test_scoring.py`